backend/src/migrations/004_comment_response_refactor.ts

```diff
@@ -0,0 +1,73 @@
+import AccessRequestModel from '../models/AccessRequest.js'
+import ReleaseModel from '../models/Release.js'
+import ResponseModel, { ResponseKind } from '../models/Response.js'
+import ReviewModel from '../models/Review.js'
+import { toEntity } from '../utils/entity.js'
+
+export async function up() {
+  // Resolve release comments
+  const releases: any = await ReleaseModel.find({})
+  for (const release of releases) {
+    release.commentIds = []
+    if (release.get('comments') !== undefined) {
+      for (const releaseComment of release.get('comments')) {
+        const newComment = new ResponseModel({
+          user: toEntity('user', releaseComment.user),
+          kind: ResponseKind.Comment,
+          comment: releaseComment.message,
+          createdAt: releaseComment.createdAt,
+        })
+        await newComment.save()
+        release.commentIds.push(newComment._id)
+        release.set('comments', undefined, { strict: false })
+      }
+      await release.save()
+    }
+  }
+
+  // Resolve access request comments
+  const accessRequests: any = await AccessRequestModel.find({})
+  for (const accessRequest of accessRequests) {
+    accessRequest.commentIds = []
+    if (accessRequest.get('comments') !== undefined) {
+      for (const accessCommment of accessRequest.get('comments')) {
+        const newComment = new ResponseModel({
+          user: toEntity('user', accessCommment.user),
+          kind: ResponseKind.Comment,
+          comment: accessCommment.message,
+          createdAt: accessCommment.createdAt,
+        })
+        await newComment.save()
+        accessRequest.commentIds.push(newComment._id)
+        accessRequest.set('comments', undefined, { strict: false })
```

Comment:
This is a dangerous operation.  Could we comment out this line initially, then add it in as a later migration once we're confident that the prior migration worked as expected?

Reply:
I've done the same for releases and also for reviews where I previously did `delete review.comments`